spring-graphql/src/main/java/org/springframework/graphql/execution/ContextDataFetcherDecorator.java

```diff
@@ -84,6 +86,24 @@ public Object get(DataFetchingEnvironment environment) throws Exception {
 		return value;
 	}
 
+	@SuppressWarnings("unchecked")
+	private Publisher<?> interceptSubscriptionPublisherWithExceptionHandler(Publisher<?> publisher) {
+		Function<? super Throwable, Mono<DataFetcherResult<?>>> onErrorResumeFunction = e ->
+				subscriptionExceptionResolver.resolveException(e)
+						.map(errors -> DataFetcherResult.newResult().errors(errors).build());
```

Comment:
I'm not sure how this even works, since it would emit a `DataFetcherResult` while the transport expects the `Publisher` to emit `ExecutionResult` instances.

I think this should return a `Mono.error` with a new exception (e.g. `SubscriptionStreamException`) that wraps and exposes the `List<GraphQLError>` through a getter. This would force the transport to terminate in the very least, but should actually handle this exception to extract the errors and create an error message.

Reply:
I just added a handler of `SubscriptionStreamException` in `GraphQlWebSocketHandler` (both `webflux` and `webmvc`). Should I add it somewhere else?